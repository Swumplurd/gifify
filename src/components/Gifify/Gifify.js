import { useState } from 'react'
import { AddCategory } from '../AddCategory/AddCategory';
import { GifGrid } from '../GifGrid/GifGrid';

export const Gifify = () => {
    const [categories, setCategories] = useState(['Demon Slayer']);

    return (
        <>
            <div className="p-5 my-4 bg-dark bg-gradient shadow-lg text-light rounded-3">
                <div className="container-fluid py-5">
                    <h1 className="display-5 fw-bold">Gifify</h1>
                    <p className="col-md-8 fs-4">Generamos peticiones http a la API de GIPHY mediante el input situado debajo... para mas informacion acerca de la API dar click en el siguiente boton...</p>
                    <a className="btn btn-primary btn-lg" target="_blank" rel="noopener noreferrer" href="https://developers.giphy.com/">API GIPHY</a>
                </div>
            </div>
            <AddCategory setCategories={setCategories}/>
            <ol>
                {
                    categories.map(category => {
                        return (
                            <GifGrid key={category} category={category}/>
                        )
                    })
                }
            </ol>
        </>
    )
}