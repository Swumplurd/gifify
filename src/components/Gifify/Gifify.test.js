import { shallow } from "enzyme"
import { Gifify } from "./Gifify"

describe('Pruebas en el componente Gifify', () => {
    test('Debe hacer match con el snapshot', () => {
        const wrapper = shallow(<Gifify />)
        expect(wrapper).toMatchSnapshot()
    })
    
})
