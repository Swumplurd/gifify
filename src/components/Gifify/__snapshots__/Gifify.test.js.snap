// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente Gifify Debe hacer match con el snapshot 1`] = `
<Fragment>
  <div
    className="p-5 my-4 bg-dark bg-gradient shadow-lg text-light rounded-3"
  >
    <div
      className="container-fluid py-5"
    >
      <h1
        className="display-5 fw-bold"
      >
        Gifify
      </h1>
      <p
        className="col-md-8 fs-4 d-none d-sm-block"
      >
        Generamos peticiones http a la API de GIPHY mediante el input situado debajo... para mas informacion acerca de la API dar click en el siguiente boton...
      </p>
      <a
        className="btn btn-primary btn-lg d-none d-sm-block"
        href="https://developers.giphy.com/"
        rel="noopener noreferrer"
        target="_blank"
      >
        API GIPHY
      </a>
    </div>
  </div>
  <AddCategory
    setCategories={[Function]}
  />
  <ol>
    <GifGrid
      category=""
      key=""
    />
  </ol>
</Fragment>
`;
